[view-transitions] Old snapshot for the root element is misplaced when the page is scrolled.

mattwoodrow opened the pull request:
#### 4c7681b6487b2aa3283a4b92ca2796874fe06f4b
<pre>
[view-transitions] Old snapshot for the root element is misplaced when the page is scrolled.
<a href="https://bugs.webkit.org/show_bug.cgi?id=274403">https://bugs.webkit.org/show_bug.cgi?id=274403</a>
&lt;<a href="https://rdar.apple.com/128406598">rdar://128406598</a>&gt;

Reviewed by Tim Nguyen.

This causes flashes on <a href="https://malcolmkee.com/blog/view-transition-api-in-react-app/#practical-use-cases">https://malcolmkee.com/blog/view-transition-api-in-react-app/#practical-use-cases</a>

When capturing the root, the snapshot is of the visible viewport area, not the whole element.
Adjust the snapshot capture rect by the scroll position to account for this.

* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/new-content-root-scrollbar-with-fixed-background-expected.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/new-content-root-scrollbar-with-fixed-background.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/old-content-root-scrollbar-with-fixed-background-expected.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/old-content-root-scrollbar-with-fixed-background.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background.html.
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::snapshotElementVisualOverflowClippedToViewport):

Canonical link: <a href="https://commits.webkit.org/279432@main">https://commits.webkit.org/279432@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb2e7ce0cd4a4dc13dc237b9833b273beb087dd8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53438 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32794 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5943 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56718 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4164 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55744 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40232 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3939 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43318 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2725 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55536 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/30986 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46176 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24456 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/27846 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3492 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2320 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/49606 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3666 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58313 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28595 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3664 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50717 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29802 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46368 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50058 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11657 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30730 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29572 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/old-content-root-scrollbar-with-fixed-background-expected.html:
```diff
@@ -0,0 +1,41 @@
+
+  <div id=container>
+   <div id=first></div>
+  </div>
```
Nit: Weird indent (compared to rest of file)

nt1m commented on Source/WebCore/dom/ViewTransition.cpp:
```diff
@@ -277,11 +277,18 @@ static RefPtr<ImageBuffer> snapshotElementVisualOverflowClippedToViewport(LocalF
+        paintRect.moveBy(scrollPosition);
+    }
+
+
```
Nit: extra blank line

mattwoodrow committed:
[view-transitions] Old snapshot for the root element is misplaced when the page is scrolled.
https://bugs.webkit.org/show_bug.cgi?id=274403
<rdar://128406598>

Reviewed by Tim Nguyen.

This causes flashes on https://malcolmkee.com/blog/view-transition-api-in-react-app/#practical-use-cases

When capturing the root, the snapshot is of the visible viewport area, not the whole element.
Adjust the snapshot capture rect by the scroll position to account for this.

* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/new-content-root-scrollbar-with-fixed-background-expected.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/new-content-root-scrollbar-with-fixed-background.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/old-content-root-scrollbar-with-fixed-background-expected.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/old-content-root-scrollbar-with-fixed-background.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/root-scrollbar-with-fixed-background.html.
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::snapshotElementVisualOverflowClippedToViewport):

Canonical link: https://commits.webkit.org/279432@main
